include/warehouse_ros/impl/query_results_impl.hpp

```diff
@@ -83,15 +83,29 @@ void ResultIterator<M>::increment()
 template <class M>
 typename MessageWithMetadata<M>::ConstPtr ResultIterator<M>::dereference() const
 {
-  ROS_ASSERT(results_);
```

Comment:
I prefer ros2 assert, because it throws an exception which is similar to the behavior of ros1. But it is only added on Foxy. I will rebase and make the change.

Reply:
Done, using `rcpputils::assert_true`.